### ISSUE TITLE
--version does not show version

### ISSUE THREAD

**Issue by sebastianbergmann** [Roave/BackwardCompatibilityCheck#157]
```
$ wget https://github.com/Roave/BackwardCompatibilityCheck/releases/download/4.0.0/roave-backward-compatibility-check.phar

$ php roave-backward-compatibility-check.phar --version
Console Tool
```

**Comment by Ocramius** [comment]
Is this a default thing of PHARs?

**Comment by sebastianbergmann** [comment]
Not sure what you mean by "default thing".  `--help` shows that `--version` is available.

**Comment by Ocramius** [comment]
Oh, interesting!

I know we hadn't built anything for showing the version, so it is probably a symfony CLI default that we didn't configure.

**Comment by asgrim** [comment]
Guess we'd just need to add these parameters:
```php
$application = new Application('Roave BackwardCompatibilityCheck', '4.0.0');
```
Though a nicer approach would be to auto determine the version... maybe using `ocramius/package-versions`? :)

**Comment by asgrim** [comment]
merged for next release (currently `5.0.0`), thanks @Ocramius

**Comment by Ocramius** [comment]
Careful with this release: `box` did run, but it also failed at something (while continuing execution):

```
Loading composer repositories with package information

Updating dependencies

Your requirements could not be resolved to an installable set of packages.

  Problem 1

    - This package requires php ^7.3 but your PHP version (7.2.0; Package overridden via config.platform (actual: 7.3.9)) does not satisfy that requirement.

  Problem 2

    - ocramius/package-versions 1.5.1 requires php ^7.3.0 -> your PHP version (7.3.9) overridden by "config.platform.php" version (7.2.0) does not satisfy that requirement.

    - ocramius/package-versions 1.5.1 requires php ^7.3.0 -> your PHP version (7.3.9) overridden by "config.platform.php" version (7.2.0) does not satisfy that requirement.

    - Installation request for ocramius/package-versions ^1.5.1 -> satisfiable by ocramius/package-versions[1.5.1].

Running update with --no-dev does not mean require-dev is ignored, it just means the packages will not be installed. If dev requirements are blocking the update you have to resolve those problems.
```

/cc @theofidry

**Comment by theofidry** [comment]
Could you print the whole stack? (use the `--debug` flag of box or `-vvv`)

**Comment by Ocramius** [comment]
It only runs on tags in CI: https://travis-ci.org/Roave/BackwardCompatibilityCheck/jobs/586453576

I think it tries to do a `composer update --no-dev` internally to remove dev deps, and fails there due due to `config.platform` provided by box.

**Comment by theofidry** [comment]
Isn't the error occurring _before_ though? Box only tries to dump the autoload (`dump-autoload`): https://github.com/humbug/box/blob/master/src/Composer/ComposerOrchestrator.php#L119. It does not try an `update` or `install`

**Comment by Ocramius** [comment]
Indeed, please disregard: error is on our side (and script is wrong).

https://github.com/Roave/BackwardCompatibilityCheck/blob/5092098994e3de84300a71c997494d6f637e839d/build-phar.sh#L7-L9

Sorry for bothering, @theofidry

**Comment by theofidry** [comment]
np, you probably should upgrade Box thought, 3.8 is out so using the 3.0 beta is a bit... risky :D

**Comment by sebastianbergmann** [comment]
This is still not fixed in 5.0.0:
```
$ wget https://github.com/Roave/BackwardCompatibilityCheck/releases/download/5.0.0/roave-backward-compatibility-check.phar                                                                         
```
```
$ php roave-backward-compatibility-check.phar --version
roave/backward-compatibility-check No version set (parsed as 1.0.0)@
```

**Comment by Ocramius** [comment]
Moved the milestone and re-opened then 👍

I wonder if this is due to no plugins running during box installation 🤔

**Comment by morozov** [comment]
This looks to be caused by https://github.com/humbug/box/issues/460. Same output:
```
$ roave-backward-compatibility-check --version

roave/backward-compatibility-check No version set (parsed as 1.0.0)@
```

**Comment by Ocramius** [comment]
This is potentially solved in #254, which uses `ocramius/package-versions:^1.10.0`, which in turn relies on `composer-runtime-api:^2`.

That way, the version should always be set.

Cons: `composer/composer:^2` will be required, but I can also see it as a plus :)
